ArchitectureFlow/ArchitectureFlow/Login/UseCase/Login.h

```diff
@@ -0,0 +1,11 @@
+#import <Foundation/Foundation.h>
+
+@class Promise;
+@class LoginRequestModel;
+@protocol LoginOutput;
+
+@protocol Login
+
+- (Promise *)executeWithRequest:(LoginRequestModel *)requestModel output:(id <LoginOutput>)output;
```

Comment:
No, that is a big no no in clean architecture. The Promise musn't be a source of our result, because there is no result as such of the use case, there is the output contract and everything that goes out goes through the output